src/bosh_azure_cpi/lib/cloud/azure/models/vm_cloud_props.rb

```diff
@@ -99,21 +102,69 @@ def _default_fault_domain_count(global_azure_config)
       end
     end
 
+    # @return [Array<Bosh::AzureCloud::LoadBalancerConfig>,nil]
     def _parse_load_balancer_config(vm_properties, global_azure_config)
-      if vm_properties[LOAD_BALANCER_KEY].is_a?(Hash)
-        resource_group_name = vm_properties[LOAD_BALANCER_KEY][RESOURCE_GROUP_NAME_KEY] || global_azure_config.resource_group_name
-        Bosh::AzureCloud::LoadBalancerConfig.new(
-          resource_group_name,
-          vm_properties[LOAD_BALANCER_KEY][NAME_KEY]
-        )
-      else
-        Bosh::AzureCloud::LoadBalancerConfig.new(
-          global_azure_config.resource_group_name,
-          vm_properties[LOAD_BALANCER_KEY]
-        )
+      load_balancer_config = vm_properties[LOAD_BALANCER_KEY]
+
+      return nil unless load_balancer_config
+
+      cloud_error("Property '#{LOAD_BALANCER_KEY}' must be a String, Hash, or Array.") unless load_balancer_config.is_a?(String) || load_balancer_config.is_a?(Hash) || load_balancer_config.is_a?(Array)
```

Comment:
Maybe we can have a generic function which parses the property. It can return a structure back like [{names => names, group_name => group_names, pool_name => pool_names}]. This output can be used to instantiate the corresponding  `Config` here. We could use this for the load balancer and application gateway. Just a suggestion.

Reply:
The 2 methods' implementations are currently slightly distinct (e.g. in the handling of `nil` vs defaulted values for the `resource_group_name`), and I'm not sure how much benefit there would currently be in consolidating only these 2 methods. If additional similar methods are needed in future, I'd agree that some de-duping via helper methods would probably be better.

Reply:
fine with me.